legion/engine/core/particles/particlesystem.cpp

```diff
@@ -0,0 +1,117 @@
+#include <core/particles/particles.hpp>
+#include <core/common/debugrendering.hpp>
+
+
+namespace legion::core
+{
+    void ParticleSystem::setup()
+    {
+        log::filter(log::severity_debug);
+        log::debug("ParticleSystem setup");
+
+        bindToEvent<events::component_creation<particle_emitter>, &ParticleSystem::emitter_setup>();
+    }
+
+    void ParticleSystem::shutdown()
+    {
+        log::debug("ParticleSystem shutdown");
+    }
+
+    void ParticleSystem::update(time::span deltaTime)
+    {
+        ecs::filter<particle_emitter> filter;
+        for (auto& ent : filter)
+        {
+            auto& emitter = ent.get_component<particle_emitter>().get();
+            if (emitter.currentParticleCount < emitter.maxSpawnCount)
+            {
+                emitter.elapsedTime += deltaTime;
```

Comment:
so the elapsedTime does not progress once they've spawned the max amount of particles?